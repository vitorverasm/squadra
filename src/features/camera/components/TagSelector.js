import React from 'react';
import { FlatList } from 'react-native';
import TagCard from './TagCard';

const TagSelector = ({ tags, currentTag, toggleTag }) => (
  <FlatList
    style={{ flex: 1, backgroundColor: 'transparent' }}
    contentContainerStyle={{ flexGrow: 0, alignItems: 'center', paddingHorizontal: 10 }}
    horizontal
    showsHorizontalScrollIndicator={false}
    data={tags}
    renderItem={({ item }) => (
      <TagCard
        tagName={item.tagName}
        onPress={() => {
          console.log('currentTag fora: ', currentTag);
          toggleTag(item, currentTag);
        }}
        checked={currentTag.id === item.id}
      />
    )}
    keyExtractor={item => item.id}
  />
);

export default TagSelector;
