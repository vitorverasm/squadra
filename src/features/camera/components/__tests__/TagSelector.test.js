import { shallow } from 'enzyme';
import React from 'react';
import TagSelector from '../TagSelector';

describe('TagSelector', () => {
  const store = {
    tags: [
      { id: '0', tagName: 'Neural Networks' },
      { id: '1', tagName: 'Statistics' },
      { id: '2', tagName: 'Linear Algebra' },
      { id: '3', tagName: 'Artificial Inteligence' }
    ],
    currentTag: { id: null, tagName: null }
  };

  it('renders correctly', () => {
    const wrapper = shallow(
      <TagSelector tags={store.tags} currentTag={store.currentTag} toggleTag={jest.fn} />
    );
    expect(wrapper.debug()).toMatchSnapshot();
  });
  it('FlatList functions', () => {
    const wrapper = shallow(
      <TagSelector tags={store.tags} currentTag={store.currentTag} toggleTag={jest.fn} />
    );
    const flatList = wrapper.find('FlatList');
    expect(flatList.props().renderItem({ item: store.tags[0] })).not.toBeNull();
    expect(flatList.props().keyExtractor({ item: store.tags[0] })).not.toBeNull();
  });
});
