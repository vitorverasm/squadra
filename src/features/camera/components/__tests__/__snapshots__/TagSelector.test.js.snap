// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagSelector renders correctly 1`] = `"<FlatList style={{...}} contentContainerStyle={{...}} horizontal={true} showsHorizontalScrollIndicator={false} data={{...}} renderItem={[Function: renderItem]} keyExtractor={[Function: keyExtractor]} disableVirtualization={false} initialNumToRender={10} maxToRenderPerBatch={10} onEndReachedThreshold={2} scrollEventThrottle={50} updateCellsBatchingPeriod={50} windowSize={21} numColumns={1} removeClippedSubviews={false} />"`;
